src/Agent.cs

```diff
@@ -26,6 +26,14 @@ public class Agent
         private readonly Collector _collector;
         private static readonly ILog _log = LogManager.GetCurrentClassLogger();
 
+        public int MessageCount
+        {
+            get
+            {
+                return _collector.MessageCount;
```

Comment:
Does C# have any concept of delegates? It would potentially make this shorter, but maybe not clearer.